src/pages/physical-count-itemDetails/index.js

```diff
@@ -0,0 +1,633 @@
+      label: _labels.metalPurity,
+      name: 'metalPurity',
+      defaultValue: 0,
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'numberfield',
+      label: _labels.weight,
+      name: 'weight',
+      defaultValue: 0,
+      props: {
+        readOnly: true
+      }
+    }
+  ]
+
+  const clearGrid = () => {
+    formik.setFieldValue('rows', formik.initialValues.rows)
+
+    setFilteredItems([])
+    setEditMode(false)
+  }
+
+  const handleClick = async dataList => {
+    try {
+      setFilteredItems([])
+      console.log('in handle click')
+      console.log(dataList)
+
+      const filteredItemsList = dataList
+        .filter(item => item.metalId && item.metalId.toString().trim() !== '')
+        .map(item => ({
+          qty: item.countedQty,
+          metalRef: null,
+          metalId: item.metalId,
+          metalPurity: item.metalPurity,
+          weight: item.weight,
+          priceType: item.priceType
+        }))
+      setFilteredItems(filteredItemsList)
+      setEditMode(dataList.length > 0)
+    } catch (exception) {}
+  }
+
+  const isPosted = formik.values.status === 3
+
+  const isHeader =
+    formik.values.stockCountId != null &&
+    formik.values.siteId != null &&
+    formik.values.controllerId != null &&
+    formik.values.SCStatus != 3 &&
+    formik.values.SCWIP != 2 &&
+    formik.values.EndofSiteStatus != 3
+
+  const isSaved =
+    formik.values.stockCountId != null &&
+    formik.values.siteId != null &&
+    formik.values.controllerId != null &&
+    formik.values.status != 3 &&
+    formik.values.SCStatus != 3 &&
+    formik.values.SCWIP != 2 &&
+    formik.values.EndofSiteStatus != 3
+
+  const onPost = async () => {
+    const status = formik.values.status == 1 ? 3 : 1
+    formik.setFieldValue('status', status)
+
+    const StockCountControllerTab = {
+      siteId: formik.values.siteId,
+      controllerId: formik.values.controllerId,
+      stockCountId: formik.values.stockCountId,
+      status: status
+    }
+
+    const res = await postRequest({
+      extension: SCRepository.StockCountControllerTab.set,
+      record: JSON.stringify(StockCountControllerTab)
+    })
+
+    if (status == 3) {
+      toast.success(platformLabels.Posted)
+    } else {
+      toast.success(platformLabels.Unposted)
+    }
+  }
+
+  const actions = [
+    {
+      key: 'Metals',
+      condition: true,
+      onClick: 'onClickMetal'
+    },
+    {
+      key: 'Locked',
+      condition: isPosted,
+      onClick: 'onUnpostConfirmation',
+      onSuccess: onPost,
+      disabled: !isHeader
+    },
+    {
+      key: 'Unlocked',
+      condition: !isPosted,
+      onClick: onPost,
+      disabled: !isHeader
+    },
+    {
+      key: 'ClearGrid',
+      condition: true,
+      onClick: 'onClearGridConfirmation',
+      onSuccess: clearGrid
+    },
+    {
+      key: 'ClearHG',
+      condition: true,
+      onClick: 'onClearAllConfirmation'
+    }
+  ]
```

Comment:
![image](https://github.com/user-attachments/assets/a1a434b6-6e33-4d31-85b5-ec8b296471af)
why are the clear buttons enabled when everything is empty?

Reply:
![image](https://github.com/user-attachments/assets/c6dafbe8-d4b7-41c3-80a0-c62dfae2461e)
and why is the clearGrid enabled when the grid is empty?